job/src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java

```diff
@@ -194,39 +195,53 @@ public WorkflowRun(WorkflowJob job, File dir) throws IOException {
             }
             setResult(Result.FAILURE);
             executionPromise.setException(x);
+            return;
         }
+        throw sleep();
     }
 
-    /**
-     * Sleeps until the run is finished, updating log messages periodically.
-     */
-    void waitForCompletion() {
```

Comment:
Package-private, and anyway the `workflow-job` plugin was not intended to expose any APIs.